components/packaging/common/load-telemetry-plugin

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env sh
+
+BASE=/var/lib/docker
+TAR_DIR=$BASE/plugins/tar
+
+# make sure that the directory exists
+mkdir -p $TAR_DIR
+
+# Create the engine UUID, if it doesn't exist
+# This will be eventually handled by the docker engine
+# ref. https://github.com/moby/moby/issues/33356
+if [ -f $BASE/engine_uuid ]; then
+    UUID=$(cat $BASE/engine_uuid)
+fi
+if [ -z ${UUID+x} ]; then
+    UUID=$(cat /proc/sys/kernel/random/uuid | tee $BASE/engine_uuid)
+fi
+
+# make sure that docker is running
+if ! printf "%s" "$DOCKER_OPTS" | grep -qE -e '-H|--host'; then
+        DOCKER_SOCKET=/var/run/docker.sock
+else
+        DOCKER_SOCKET=$(printf "%s" "$DOCKER_OPTS" | grep -oP -e '(-H|--host)\W*unix://\K(\S+)' | sed 1q)
```

Comment:
@thaJeztah I think it will never happen at `installation` time. But for the cases you described, `DOCKER_SOCKET` will be empty and the `script` will skip this part. We should `exit` after that.